### DISCUSSION TITLE
Why are updates only collected within one commit when the 'canary' option is set to true?

### DISCUSSION THREAD

**Discussion by goumang2010** [ghiscoding/lerna-lite#748] [General]
why using  ^..  in line 70

expect:  committish = `${sha}^`

packages/core/src/utils/collect-updates/collect-updates.ts

<img width="840" alt="image" src="https://github.com/lerna-lite/lerna-lite/assets/12218607/3c5af9f9-15a9-4115-94bd-a53f2384e854">

**Comment by ghiscoding** [comment]
Sorry but I don't know, that code was created long time ago by the original Lerna author and is not something that I don't fully understand. However it does say that it will return anything in between, which could be 1 or many as written in the comments. I would suggest to always search in the original Lerna project for issues, but I didn't find much to be related, there are issue [2060](https://github.com/lerna/lerna/issues/2060) and maybe [1893](https://github.com/lerna/lerna/issues/1893#issuecomment-602691974) but I think these are different type of problems. Sorry I can't help much more than that, I don't have full knowledge of its core code
